Source/MARS.Client.Client.pas

```diff
@@ -319,6 +320,7 @@ class function TMARSCustomClient.GetAsString(const AEngineURL, AAppName,
 
         LFinalURL := LResource.URL;
         LResource.SpecificToken := AToken;
+        LResource.SpecificAccept := AAccept;

```

Comment:
Default Accept for MARS client is `TMediaType.APPLICATION_JSON`:

https://github.com/andrea-magni/MARS/blob/8ad4b364cb85def0d1acc07e73c6e1b5b3a63303/Source/MARS.Client.Application.pas#L79-L87

https://github.com/andrea-magni/MARS/blob/8ad4b364cb85def0d1acc07e73c6e1b5b3a63303/Source/MARS.Client.CustomResource.pas#L542-L547

https://github.com/andrea-magni/MARS/blob/8ad4b364cb85def0d1acc07e73c6e1b5b3a63303/Source/MARS.Client.Client.pas#L308-L322

Reply:
Yeep! I find that:

https://github.com/andrea-magni/MARS/blob/8ad4b364cb85def0d1acc07e73c6e1b5b3a63303/Source/MARS.Client.CustomResource.pas#L226

@andrea-magni: Why `FSpecificAccept := TMediaType.WILDCARD;` by default but `function TMARSClientCustomResource.GetAccept` is setting this to `Application.DefaultMediaType` which is `TMediaType.APPLICATION_JSON`. Is this consistent?

Reply:
TMARSCustomClientResource  is a base class for resources, specific subclasses will have a different default for SpecificAccept (see https://github.com/andrea-magni/MARS/blob/master/Source/MARS.Client.Resource.JSON.pas#L119).
A resource with no specific accept value, will default to the default of its application.

Reply:
The TMediaType.WILDCARD seems a good default for the TMARSCustomClient.GetAsString as it is very high level (it could be used to retrieve JSON as well as XML, CSV or other text-based formats...). We are talking about TMARSCustomClient.GetAsString (a class function meant to be used apart from the component-based approach) and not the TMARSClientCustomResource.GetAsString.

Reply:
OK! Now I understand. Thanks for the explanation. I'm OK with this code.

For the future: It's not so ease to understand. Maybe the code should be a little cleaner. :-) Single Responsibility Principle? Maybe it such helper methods should be extracted to different class / unit, decoupled from component declaration?

Reply:
On my todo list there is a heavy refactoring of the MARS client library with respect to the non-component based approach. I'd like something with a fluent syntax style but it is out of range at the moment! :-)